File: src/searchingAlgorithms/LinearSearch.java
package searchingAlgorithms;

public class LinearSearch {

    public static void main(String[] arg){
        //create dat set for 1 million
        DataSet dataSet = new DataSet(10000000);


        int search = 9999999;

        for (int i = 0; i < dataSet.data.length; i++) {
            //number try
            dataSet.numberOfTry++;

            //check it
            if (dataSet.data[i] == search){
                System.out.println("Element is found on" +
                        " after "+dataSet.numberOfTry+" try");
                break;
            }
        }

    }

    // so here big o is n
}

File: src/searchingAlgorithms/LinearSearch.kt
package searchingAlgorithms

fun main(args: Array<String>) {

    val data = DataK(100000)

    val search = 99999;
    var numberOfTry = 0

    var result = - 1


    for (i in 0 until (data.array.size - 1)) {
        val number = data.array[i]
        if (number == search) {
            result = number as Int
            break
        }

        numberOfTry++
    }

    if (result != -1)
        println("Data found on: $numberOfTry")
    else
        println("Data not found try: $numberOfTry")
}